synapse/federation/federation_server.py

```diff
@@ -360,13 +359,13 @@ async def _handle_incoming_transaction(
         # want to block things like to device messages from reaching clients
         # behind the potentially expensive handling of PDUs.
         pdu_results, _ = await make_deferred_yieldable(
-            defer.gatherResults(
-                [
+            gather_results(
+                (
```

Comment:
If I understand rightly, the difference here is that we're returning a Tuple (which we can annotate heterogeneously) as opposed to a list? But otherwise `pdu_results, _` is identical

Reply:
Yes, the values of `pdu_results, _` should be the same before and after this PR.